src/frontend/src/icp-eth/components/tokens/AddTokenByNetwork.svelte

```diff
@@ -0,0 +1,40 @@
+<script lang="ts">
+	import { i18n } from '$lib/stores/i18n.store';
+	import { Dropdown, DropdownItem } from '@dfinity/gix-components';
+	import { networksMainnets } from '$lib/derived/networks.derived';
+	import { ETHEREUM_NETWORK_SYMBOL, ICP_NETWORK_SYMBOL } from '$env/networks.env';
+	import IcAddTokenForm from '$icp/components/tokens/IcAddTokenForm.svelte';
+	import AddTokenForm from '$eth/components/tokens/AddTokenForm.svelte';
+	import { networkId } from '$lib/derived/network.derived';
+	import { isNullish } from '@dfinity/utils';
+
+	export let dropdownNetwork: string | undefined;
+	export let tokenData: Record<string, string>;
+
+	let ledgerCanisterId: string = '';
+	let indexCanisterId: string = '';
+	let contractAddress: string = '';
+
+	$: tokenData = {
+		ledgerCanisterId,
+		indexCanisterId,
+		contractAddress
+	};
+</script>
+
+{#if isNullish($networkId)}
+	<div class="stretch pt-8">
+		<label for="selectedNetwork" class="font-bold px-4.5">{$i18n.tokens.manage.text.select_network}:</label>
+		<Dropdown name="selectedNetwork" bind:selectedValue={dropdownNetwork}>
+			{#each $networksMainnets as network}
+				<DropdownItem value={network.id.description ?? ''}>{network.name}</DropdownItem>
+			{/each}
+		</Dropdown>
+	</div>
+{/if}
+
+{#if dropdownNetwork === ICP_NETWORK_SYMBOL}
```

Comment:
Have you try to add a `in:fade` effect here? Does it providers a smoother experience when the network is modified?

Reply:
good point! included